files/shaders/objects_vertex.glsl

```diff
@@ -50,12 +50,16 @@ varying vec3 passNormal;
 void main(void)
 {
     gl_Position = gl_ModelViewProjectionMatrix * gl_Vertex;
-    depth = gl_Position.z;
-
     vec4 viewPos = (gl_ModelViewMatrix * gl_Vertex);
     gl_ClipVertex = viewPos;
     vec3 viewNormal = normalize((gl_NormalMatrix * gl_Normal).xyz);
 
+#if !@accurateFog
+    depth = gl_Position.z;
+#else
+    depth = length(viewPos.xyz);
```

Comment:
Are we 100% sure that `viewPos` will always be in Cartesian space, or might it sometimes be in homogeneous space? If so, `length(viewPos.xyz / viewPos.w);` would be preferable.